libr/io/io.c

```diff
@@ -519,6 +544,39 @@ R_API bool r_io_write_at(RIO* io, ut64 addr, const ut8* buf, int len) {
 	return ret;
 }
 
+// Prefix mode
+R_API int r_io_prefix_write_at(RIO *io, ut64 addr, ut8 *buf, int len) {
+	if (!io || !buf || len < 0) {
+		return -EINVAL;
+	}
+	ut8 *buf1 = (ut8 *)buf;
+	int ret;
+	if (io->write_mask) {
```

Comment:
this writemask thing shouldnt be dupped, better reuse r_io_write_at for this. 

a much simpler solution would be to add a var inside io that holds the number of bytes read/write. this way we dont need to duplicate code and we can remove the fffff hack and the isvalid+read pattern